packages/expect/src/toThrowMatchers.ts

```diff
@@ -324,13 +335,44 @@ const toThrow = (
 const formatExpected = (label: string, expected: unknown) =>
   label + printExpected(expected) + '\n';
 
-const formatReceived = (label: string, thrown: Thrown | null, key: string) => {
+const formatReceived = (
+  label: string,
+  thrown: Thrown | null,
+  key: string,
+  expected?: string | RegExp,
+) => {
   if (thrown === null) {
     return '';
   }
 
   if (key === 'message') {
-    return label + printReceived(thrown.message) + '\n';
+    const message = thrown.message;
+
+    if (typeof expected === 'string') {
+      const index = message.indexOf(expected);
+      if (index !== -1) {
+        return (
+          label +
+          printReceivedStringContainExpectedSubstring(
+            message,
+            index,
+            expected.length,
+          ) +
+          '\n'
+        );
+      }
+    } else if (expected !== undefined) {
```

Comment:
`instanceof RegExp`?

Reply:
Good question, as usual.

Some `expect` code tries to avoid `instanceof` but is it still a problem?

Reply:
I think that might be due to e.g. `Error` being weird across sandbox boundaries. But here we want you to pass something to us that we run.

I confused myself with that sentence 😛 Did it make sense?

Reply:
matcher error if not `string` or `regexp`? maybe next major

Reply:
A place to change in next major is in `createMatcher` function, replace:

```js
    } else if (expected !== null && typeof expected.test === 'function') {
      return toThrowExpectedRegExp(matcherName, options, thrown, expected);
```

with

```js
    } else if (expected instanceof RegExp) {
      return toThrowExpectedRegExp(matcherName, options, thrown, expected);
```